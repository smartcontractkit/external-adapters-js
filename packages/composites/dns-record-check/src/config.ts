import DNS from '@chainlink/dns-query-adapter'
import { Config } from '@chainlink/types'

export const makeConfig = (): Config => {
  return DNS.makeConfig()
}
