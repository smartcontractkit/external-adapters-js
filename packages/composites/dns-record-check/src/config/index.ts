import * as DNS from '@chainlink/dns-query-adapter'
import { Config } from '@chainlink/ea-bootstrap'

export const NAME = 'DNS_RECORD_CHECK'
export const DEFAULT_ENDPOINT = 'dnsQuery'

export const makeConfig = (): Config => {
  return {
    ...DNS.makeConfig(),
    defaultEndpoint: DEFAULT_ENDPOINT,
  }
}
