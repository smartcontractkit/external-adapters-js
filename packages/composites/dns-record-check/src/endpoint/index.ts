import * as dnsQuery from './dnsQuery'

export type TInputParameters = dnsQuery.TInputParameters

export * as dnsQuery from './dnsQuery'
