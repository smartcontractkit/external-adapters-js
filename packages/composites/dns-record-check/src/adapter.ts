import { ExecuteWithConfig, ExecuteFactory, Config } from '@chainlink/types'
import { Validator } from '@chainlink/ea-bootstrap'
import { Requester } from '@chainlink/ea-bootstrap'
import DNS from '@chainlink/dns-query-adapter'
import { DNSQueryResponse, DNSAnswer } from '@chainlink/dns-query-adapter/dist/types'
import { makeConfig } from './config'

const inputParams = {
  record: true,
}

const execute: ExecuteWithConfig<Config> = async (input, config) => {
  const validator = new Validator(input, inputParams)
  if (validator.error) throw validator.error

  const jobRunID = validator.validated.id
  const { record } = validator.validated.data

  const dnsExecute = DNS.makeExecute(config)
  const dnsResponse = await dnsExecute(input)
  const dnsData: DNSQueryResponse = { ...dnsResponse.data }
  const foundRecord = dnsData.Answer.find((ans: DNSAnswer) => ans.data.includes(record))

  return Requester.success(
    jobRunID,
    {
      status: 200,
      data: {
        result: !!foundRecord,
      },
    },
    config.verbose,
  )
}

export const makeExecute: ExecuteFactory<Config> = (config?: Config) => (input) =>
  execute(input, config || makeConfig())
