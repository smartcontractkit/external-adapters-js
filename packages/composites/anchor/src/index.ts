import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute } from './adapter'
import { makeConfig, NAME } from './config'

/**
 * NOTE: This adapter could probably be made more general in the future.  All it is really doing is fetching an intermediary token price
 * using the Token Allocation adapter and handling the price after that.
 */

const adapterContext = { name: NAME }

const { server } = expose(adapterContext, makeExecute(), undefined, endpointSelector)

export { NAME, makeConfig, makeExecute, server }
