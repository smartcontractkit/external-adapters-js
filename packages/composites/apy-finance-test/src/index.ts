import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
import { Adapter } from '@chainlink/external-adapter-framework/adapter'
import { allocations, tvl } from './endpoint'
import { config } from './config'

export const adapter = new Adapter({
  defaultEndpoint: tvl.name,
  name: 'APY_FINANCE_TEST',
  config,
  endpoints: [allocations, tvl],
})

export const server = (): Promise<ServerInstance | undefined> => expose(adapter)
