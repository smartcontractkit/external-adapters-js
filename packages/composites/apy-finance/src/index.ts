import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { makeConfig } from './config'

const NAME = 'APY-Finance'

export = { NAME, makeConfig, makeExecute, ...expose(NAME, makeExecute()) }
