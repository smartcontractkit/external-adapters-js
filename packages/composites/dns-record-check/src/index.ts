import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { NAME } from './config'

export = { makeExecute, ...expose(NAME, makeExecute()) }
