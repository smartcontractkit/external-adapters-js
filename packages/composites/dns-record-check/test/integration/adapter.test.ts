import { AdapterRequest } from '@chainlink/ea-bootstrap'
import * as process from 'process'
import { server as startServer } from '../../src'
import { mockRecordCheckResponse } from './fixtures'
import { setupExternalAdapterTest } from '@chainlink/ea-test-helpers'
import type { SuiteContext } from '@chainlink/ea-test-helpers'
import { SuperTest, Test } from 'supertest'

describe('dns record check', () => {
  const context: SuiteContext = {
    req: null,
    server: startServer,
  }

  const envVariables = {
    CACHE_ENABLED: 'false',
    DNS_PROVIDER: process.env.DNS_PROVIDER || 'google',
  }

  setupExternalAdapterTest(envVariables, context)

  describe('record check endpoint', () => {
    const recordCheckRequest: AdapterRequest = {
      id: '1',
      data: {
        name: 'example.com',
        type: 'TXT',
      },
    }

    it('should return success', async () => {
      mockRecordCheckResponse()

      const response = await (context.req as SuperTest<Test>)
        .post('/')
        .send(recordCheckRequest)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
