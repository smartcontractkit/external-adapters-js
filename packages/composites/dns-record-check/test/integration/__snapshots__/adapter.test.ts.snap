// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dns record check record check endpoint should return success 1`] = `
{
  "data": {
    "result": true,
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": true,
  "statusCode": 200,
}
`;
